### ISSUE TITLE
[BUG] FileInput cancel operation throws error

### ISSUE THREAD

**Issue by jeffreymonaco** [bokeh/bokeh#10887]
**SUMMARY:** FileInput widget throws an error if the user cancels the input. NB: Requires a previously successfully executed file-input operation for error to manifest.

**VERSIONS:** Bokeh 2.2.3, Python 3.8.5, Mac OSX Big Sur

**Minimal Reproducible Example**

```
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
"""
from bokeh.models import FileInput
from bokeh.io import curdoc

def file_input_cb(attr, old, new):
    if new:
        print("INFO attr {:} old {:} new {:}".format(attr, old, new))

ui = FileInput(accept='.csv,.json.,.txt', multiple=False)
ui.on_change('value', file_input_cb)

curdoc().add_root(ui)

```

**STACK TRACE**

```
2021-02-03 10:07:00,217 Starting Bokeh server version 2.2.3 (running on Tornado 6.1)
2021-02-03 10:07:00,218 User authentication hooks NOT provided (default user enabled)
2021-02-03 10:07:00,220 Bokeh app running at: http://localhost:5006/file_inp
2021-02-03 10:07:00,220 Starting Bokeh server with process id: 49395
2021-02-03 10:07:28,470 WebSocket connection opened
2021-02-03 10:07:28,470 ServerConnection created
INFO attr value old  new TU5WUl9BCk1OVlJfQgpNTlZSX0MKTU5WUl9ECk1OVlJfRQpNTlZSX0YKTU5WUl9HCk1OVlJfSApNTlZSX0kKTU5WUl9KCk1OVlJfSwpNTlZSX0wKTU5WUl9NCk1OVlJfTgpNTlZSX08KTU5WUl9QCk1OVlJfUQpNTlZSX1IKTU5WUl9TCk1OVlJfVApNTlZSX1UKTU5WUl9WCk1OVlJfVwpNTlZSX1gKTU5WUl9ZCk1OVlJfWgo=
2021-02-03 10:07:47,776 error handling message
 message: Message 'PATCH-DOC' content: {'events': [{'kind': 'ModelChanged', 'model': {'id': '1001'}, 'attr': 'filename'}], 'references': []} 
 error: KeyError('new')
Traceback (most recent call last):
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/server/protocol_handler.py", line 90, in handle
    work = await handler(message, connection)
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/server/session.py", line 67, in _needs_document_lock_wrapper
    result = func(self, *args, **kwargs)
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/server/session.py", line 261, in _handle_patch
    message.apply_to_document(self.document, self)
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/protocol/messages/patch_doc.py", line 100, in apply_to_document
    doc._with_self_as_curdoc(lambda: doc.apply_json_patch(self.content, setter))
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/document/document.py", line 1169, in _with_self_as_curdoc
    return f()
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/protocol/messages/patch_doc.py", line 100, in <lambda>
    doc._with_self_as_curdoc(lambda: doc.apply_json_patch(self.content, setter))
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/document/document.py", line 409, in apply_json_patch
    value = event_json['new']
KeyError: 'new'
2021-02-03 10:07:47,779 error handling message
 message: Message 'PATCH-DOC' content: {'events': [{'kind': 'ModelChanged', 'model': {'id': '1001'}, 'attr': 'mime_type'}], 'references': []} 
 error: KeyError('new')
Traceback (most recent call last):
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/server/protocol_handler.py", line 90, in handle
    work = await handler(message, connection)
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/server/session.py", line 67, in _needs_document_lock_wrapper
    result = func(self, *args, **kwargs)
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/server/session.py", line 261, in _handle_patch
    message.apply_to_document(self.document, self)
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/protocol/messages/patch_doc.py", line 100, in apply_to_document
    doc._with_self_as_curdoc(lambda: doc.apply_json_patch(self.content, setter))
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/document/document.py", line 1169, in _with_self_as_curdoc
    return f()
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/protocol/messages/patch_doc.py", line 100, in <lambda>
    doc._with_self_as_curdoc(lambda: doc.apply_json_patch(self.content, setter))
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/document/document.py", line 409, in apply_json_patch
    value = event_json['new']
KeyError: 'new'
2021-02-03 10:07:47,780 error handling message
 message: Message 'PATCH-DOC' content: {'events': [{'kind': 'ModelChanged', 'model': {'id': '1001'}, 'attr': 'value'}], 'references': []} 
 error: KeyError('new')
Traceback (most recent call last):
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/server/protocol_handler.py", line 90, in handle
    work = await handler(message, connection)
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/server/session.py", line 67, in _needs_document_lock_wrapper
    result = func(self, *args, **kwargs)
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/server/session.py", line 261, in _handle_patch
    message.apply_to_document(self.document, self)
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/protocol/messages/patch_doc.py", line 100, in apply_to_document
    doc._with_self_as_curdoc(lambda: doc.apply_json_patch(self.content, setter))
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/document/document.py", line 1169, in _with_self_as_curdoc
    return f()
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/protocol/messages/patch_doc.py", line 100, in <lambda>
    doc._with_self_as_curdoc(lambda: doc.apply_json_patch(self.content, setter))
  File "/Users/x/opt/anaconda3/lib/python3.8/site-packages/bokeh/document/document.py", line 409, in apply_json_patch
    value = event_json['new']
KeyError: 'new'
```

**Comment by bryevdv** [comment]
@jeffreymonaco Do you have any more information that might be relevant (browser?) I cannot reproduce any issue with the code above on OSX with either 2.2.3 or current `branch-2.3` tip. I did make sure to complete a selection first. Perhaps you can record a screengrab?

**Comment by jeffreymonaco** [comment]
The tests that produced the key error in the server terminal winder were run using Google Chrome Version 88.0.4324.150 (Official Build) (x86_64). The server was running on OS X Big Sur and the client browser was run on the same computer, accessing the server via localhost.

**Comment by pvillacorta-stratio** [comment]
I am getting the same error when I click the button and then press ESC, after a previous successful file input.

**VERSIONS**: Bokeh 2.2.2, Python 3.7.6, Ubuntu 18.04.5 LTS

**Comment by mattpap** [comment]
On bokeh 2.3.0 in JS console:
```
Uncaught (in promise) Error: FileInput(1013).value given invalid value: undefined
    at PrimitiveProperty.validate (properties.js:105)
    at PrimitiveProperty._update (properties.js:91)
    at PrimitiveProperty.set_value (properties.js:78)
    at FileInput._setv (has_props.js:263)
    at FileInput.setv (has_props.js:303)
    at FileInput.set [as value] (has_props.js:104)
    at FileInputView.load_files (file_input.js:48)
```
No errors in the terminal.

**Comment by bryevdv** [comment]
This landed on `branch-2.4`  but I intend to backport it for a 2.3.1 release
